src/http/server_context.rs

```diff
@@ -23,6 +25,7 @@ pub struct ServerContext {
   pub gql_data_loaders: Arc<Vec<DataLoader<DataLoaderRequest, GraphqlDataLoader>>>,
   pub cache: ChronoCache<u64, ConstValue>,
   pub grpc_data_loaders: Arc<Vec<DataLoader<grpc::DataLoaderRequest, GrpcDataLoader>>>,
+  pub num_requests_fetched: Arc<Mutex<HashMap<String, NumRequestsFetched>>>,
```

Comment:
Use a trait that does rate limit. That looks something like —

```rust
trait RateLimiter<K> {
  async fn allow(key: K) -> Result<Stats, Error>
}

struct Stats {
  bucket_size: usize,
  filled: usize,
}

```

Reply:
Instead of a trait, I have created a `struct RateLimiter` for now, since in this change there is only one type of rate limiter, once another one is added I will extract out the common elements from both in a `trait` as you have suggested